### ISSUE TITLE
Significant slowdown between 1.2.7–1.2.9.10

### ISSUE THREAD

**Issue by patricklucas** [yahoo/kafka-manager#160]
I recently tried to upgrade from 1.2.7 to 1.2.9.10, but unfortunately the latest version is unusable and all useful operations on a cluster with a significant number of topics all result in this error now:

```
Yikes! Ask timed out on [ActorSelection[Anchor(akka://kafka-manager-system/), Path(/user/kafka-manager/<cluster>/kafka-state)]] after [2000 ms]
```

I tried switching back and forth a number of times, and the issue reproduces reliably.

kafka-manager still works on my smaller clusters, with topic counts in the low hundreds. But any cluster with a few hundred to a few thousand topics time out on 100% of requests.

The only solution currently appears to be to stay on 1.2.7, but you might want to put a bit of focus on performance to ensure this great tool remains useful for production workloads.

**Comment by patelh** [comment]
HI Patrick, thank you for the feedback.  The consumer offset polling needs to be less aggressive.  I am working on that refactor right now.  Can you try by enabling active offset cache in the cluster config?

**Comment by patricklucas** [comment]
The configuration option for that claims "Not recommended for large # of consumers", which we do. What's the impact? Just more ZK watchers?

**Comment by patelh** [comment]
Yes, but you can just wait till next week and I'll have the refactor done.  @jisookim0513 is also going to add option to disable consumer offset polling.  The next thing will be to add kafka based offset checking.  Right now it is just zk based so if all your consumers are ZK based offset tracking then it could a lot of updates for KM to process.

**Comment by patelh** [comment]
Most of the changes are in master now.  Just need to get the upgrade to play 2.4 tested and merged before tagging.

**Comment by patricklucas** [comment]
Great, once you cut a release I'll give it a shot.

**Comment by denisgarci** [comment]
Hi, I'm having the same issue after a clean install of version 1.2.9.14. Is there a fix? Thank you!

**Comment by patelh** [comment]
1.3.0.4 has been released.  Please give it a try.  There is a hard java 8 requirement now.

**Comment by denisgarci** [comment]
Great I will thank you!

**Comment by williamsjj** [comment]
@denisgarci Did 1.3.x fix the issue for you?

**Comment by denisgarci** [comment]
Yes it did! Thank you.

**Comment by divyanshi2** [comment]
My kafka manager version is 1.3.3.4 but I am still having same issue. There are 3 cluster in my kafka manager. Restarting the kafka manager fix the issue for 5-10 mins but again it reappears.
